### PR TITLE
networkd: make IPv6 route preference configurable

### PR DESCRIPTION
**Pull request by ssahani**
The work supports route preference configurable.
i.e.  able to set low, medium and high.

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -832,6 +832,16 @@
           </listitem>
         </varlistentry>
         <varlistentry>
+          <term><varname>Preference=</varname></term>
+          <listitem>
+            <para>Specifies the route preference as defined in <ulink
+            url="https://tools.ietf.org/html/rfc4191">RFC4191</ulink> for Router Discovery messages.
+            Which can be <literal>low</literal> the route has a lowest priority,
+            <literal>medium</literal> the route has a default priority or
+            <literal>high</literal> the route has a highest priority.</para>
```
Comment (fsateler) [inline man/systemd.network.xml]: "Which can be" feels off. I suggest changing to "Value can be one of". Also, I don't think the description of each value adds much, as its pretty obvious that "low" means low priority. I'd instead only note that medium is the default: "Value can be one of low, medium or high. The default setting as per the RFC is medium".

**Comment by ssahani** [comment]
updated .

File: man/systemd.network.xml [hunk 2]
```diff
@@ -842,6 +842,15 @@
           </listitem>
         </varlistentry>
         <varlistentry>
+          <term><varname>Preference=</varname></term>
```
Comment (poettering) [inline man/systemd.network.xml]: If this is specific to IPv6, I'd prefer if we'd prefix this with "IPv6", so that this becomes `IPv6Preference=`...

**Comment by ssahani** [comment]
updated .

File: src/network/networkd-route.c [hunk 3]
```diff
@@ -975,6 +977,35 @@ int config_parse_gateway_onlink(const char *unit,
...
+                                       const char *section,
+                                       unsigned section_line,
+                                       const char *lvalue,
+                                       int ltype,
+                                       const char *rvalue,
+                                       void *data,
+                                       void *userdata) {
+        Network *network = userdata;
+        _cleanup_route_free_ Route *n = NULL;
+        int r;
```
Comment (evverx) [inline src/network/networkd-route.c:997]: This causes the following warnings:
```sh
[771/1780] Compiling c object 'src/network/networkd-core@sta/networkd-route.c.o'
../src/network/networkd-route.c: In function ‘config_parse_ipv6_route_preference’:
../src/network/networkd-route.c:997:13: warning: unused variable ‘r’ [-Wunused-variable]
         int r;
             ^
../src/network/networkd-route.c:995:18: warning: unused variable ‘network’ [-Wunused-variable]
         Network *network = userdata;
                  ^~~~~~~
```

**Comment by ssahani** [comment]
Added a pr already

**Pull request merged**